src/core/namespace.c

```diff
@@ -287,10 +288,18 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                // mount() always follow symlinks, so we follow symlinks during the stat() as well
```

Comment:
FYI, I just found out via #567 / [Debian bug #792187](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=792187) that some non-vanilla kernels (Chromium kernel) can restrict the mount() syscall from following symlinks. See https://lwn.net/Articles/567871/

I don't know if that concerns us in this PR.